src/test/java/bio/terra/app/controller/AdminApiControllerTest.java

```diff
@@ -0,0 +1,155 @@
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.autoconfigure.web.servlet.WebMvcTest;
+import org.springframework.boot.test.mock.mockito.MockBean;
+import org.springframework.test.context.ActiveProfiles;
+import org.springframework.test.context.ContextConfiguration;
+import org.springframework.test.web.servlet.MockMvc;
+
+@ActiveProfiles({"google", "unittest"})
+@ContextConfiguration(classes = {AdminApiController.class, GlobalExceptionHandler.class})
+@Tag(Unit.TAG)
+@WebMvcTest
+class AdminApiControllerTest {
+
+  @Autowired private MockMvc mvc;
+
+  @MockBean private JobService jobService;
+  @MockBean private AuthenticatedUserRequestFactory authenticatedUserRequestFactory;
+  @MockBean private DrsService drsService;
+  @MockBean private IamService iamService;
+  @MockBean private DatasetService datasetService;
+  @MockBean private SnapshotService snapshotService;
+  @MockBean private ApplicationConfiguration applicationConfiguration;
+
+  private static final AuthenticatedUserRequest TEST_USER =
+      AuthenticationFixtures.randomUserRequest();
+  private static final UUID MODEL_ID = UUID.randomUUID();
+  private static final String ADMIN_DATASETS_ENDPOINT = "/api/admin/v1/datasets/{id}";
+  private static final String ADMIN_SNAPSHOTS_ENDPOINT = "/api/admin/v1/snapshots/{id}";
+  private static final IamForbiddenException FORBIDDEN_EXCEPTION =
+      new IamForbiddenException("Forbidden");
+
+  @BeforeEach
+  void setUp() {
+    when(authenticatedUserRequestFactory.from(any())).thenReturn(TEST_USER);
+  }
+
+  @Test
+  void testAdminRetrieveDataset() throws Exception {
+    when(iamService.isResourceTypeAdminAuthorized(any(), any(), any())).thenReturn(true);
+    when(datasetService.retrieveDatasetModel(any(), any(), any()))
+        .thenReturn(new DatasetModel().id(MODEL_ID));
+    String json =
+        mvc.perform(get(ADMIN_DATASETS_ENDPOINT, MODEL_ID))
+            .andExpect(status().isOk())
+            .andReturn()
+            .getResponse()
+            .getContentAsString();
+    DatasetModel model = TestUtils.mapFromJson(json, DatasetModel.class);
+    assertThat("Dataset Model is returned", model, notNullValue());
+    assertThat("Dataset Model ID is correct", model.getId(), equalTo(MODEL_ID));
+  }
+
+  @Test
+  void testAdminRetrieveDatasetInvalidId() throws Exception {
+    mvc.perform(get(ADMIN_DATASETS_ENDPOINT, "not a UUID"))
+        .andExpect(status().isBadRequest())
+        .andReturn()
+        .getResponse()
+        .getStatus();
+  }
+
+  @Test
+  void testAdminRetrieveDatasetNotAuthorized() throws Exception {
+    doThrow(FORBIDDEN_EXCEPTION)
+        .when(iamService)
+        .verifyResourceTypeAdminAuthorized(any(), any(), any());
+    int status =
+        mvc.perform(get(ADMIN_DATASETS_ENDPOINT, MODEL_ID)).andReturn().getResponse().getStatus();
+    assertThat("", status, equalTo(FORBIDDEN_EXCEPTION.getStatusCode().value()));
```

Comment:
If you don't have a message for an assertion, you can omit the first argument
```suggestion
    assertThat(status, equalTo(FORBIDDEN_EXCEPTION.getStatusCode().value()));
```

Also, this can be done using `.andExpect(status().isForbidden())`